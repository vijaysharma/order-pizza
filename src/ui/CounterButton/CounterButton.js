import React from 'react';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faMinusCircle, faPlusCircle} from '@fortawesome/free-solid-svg-icons';
import * as classes from './CounterButton.module.css';

const CounterButton = props => (
  <div className={classes.CounterButton}>
    <FontAwesomeIcon icon={faMinusCircle} onClick={props.decrement} className={[props.disabled ? classes.DecrementDisabled : '', classes.Decrement].join(' ')} />
    <span>{props.count}</span>
    <FontAwesomeIcon icon={faPlusCircle} onClick={props.increment} className={classes.Increment} />
  </div>
);


export default CounterButton;