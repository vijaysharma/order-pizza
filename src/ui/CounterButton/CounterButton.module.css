.CounterButton {
  color: #9b9b9f;
  display: inline-block;
}
.CounterButton svg {
  vertical-align: middle;
}
.CounterButton span {
  font-size: 1rem;
  padding: 0 0.75rem;
  margin: 0 0.75rem;
  display: inline-block;
  vertical-align: middle;
  width: 2.5rem;
  text-align: center;
}
.Increment, .Decrement {
  cursor: pointer;
  font-size: 2rem;
}
.Increment {
  color: #e13b5b;
}
.Decrement {
  color: #283688;
}
.DecrementDisabled {
  color: #8b8b8b;
  cursor: not-allowed;
}