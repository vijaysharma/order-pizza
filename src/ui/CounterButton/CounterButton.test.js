import React from 'react';

import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import CounterButton from './CounterButton';

configure({adapter: new Adapter()});

describe('<CounterButton/>', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<CounterButton />);
  });

  it('should render <CounterButton/> component', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('should render <CounterButton/> correctly', () => {
    wrapper.setProps({
      count: 10
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render <CounterButton/> component with count 10', () => {
    wrapper.setProps({
      count: 10
    });
    expect(wrapper.find('span').text().includes(10)).toBe(true);
  });

  it('should render <CounterButton/> component with count 5', () => {
    wrapper.setProps({
      count: 5
    });
    expect(wrapper.find('span').text().includes(5)).toBe(true);
  });

  it('should render <CounterButton/> component with icon', () => {
    expect(wrapper.find(FontAwesomeIcon)).toHaveLength(2);
  });

});