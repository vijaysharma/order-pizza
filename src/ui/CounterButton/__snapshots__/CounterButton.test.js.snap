// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CounterButton/> should render <CounterButton/> correctly 1`] = `ShallowWrapper {}`;
